src/main/java/uk/gov/hmcts/reform/ccd/documentam/service/impl/DocumentManagementServiceImpl.java

```diff
@@ -196,14 +204,27 @@ private UpdateDocumentsCommand prepareRequestForAttachingDocumentToCase(CaseDocu
         List<DocumentUpdate> documentsList = new ArrayList<>();
         for (DocumentHashToken documentHashToken : caseDocumentsMetadata.getDocumentHashTokens()) {
 
+            Optional<ResponseEntity<StoredDocumentHalResource>> documentMetadata = Optional.ofNullable(
+                getDocumentMetadata(UUID.fromString(documentHashToken.getId())));
+
             if (documentHashToken.getHashToken() != null) {
                 String hashcodeFromStoredDocument = generateHashToken(UUID.fromString(documentHashToken.getId()));
                 if (!hashcodeFromStoredDocument.equals(documentHashToken.getHashToken())) {
                     throw new ForbiddenException(UUID.fromString(documentHashToken.getId()));
                 }
+            // Token is not provided by CCD
             } else if (hashCheckEnabled) {
                 throw new ForbiddenException("Hash check is enabled but hashToken hasn't provided for the document:"
                                                  + documentHashToken.getId());
+            } else {
+                // document metadata does not exist and document is not a moving case
+                if (documentMetadata.isPresent()
+                    && !documentMetadata.get().getBody().getMetadata().isEmpty()
+                    && !isDocumentMovingCases(documentMetadata.get().getBody().getMetadata().get(CASE_TYPE_ID))) {
+                    throw new BadRequestException(String.format(
+                        "Document is not a moving case: %s",
```

Comment:
Please specify full message like.. Document metadata exists but case type is not a moving case type.